Separate interactive and non-interactive sessions

pakrym-oai opened the pull request:
Do not show exec session in VSCode/TUI selector.

pakrym-oai committed:
Separate interactive and non-interactive sessions

pakrym-oai committed:
Merge branch 'main' into pakrym/separate-interactive-and-non-interactive-sessions

# Conflicts:
#	codex-rs/cli/src/proto.rs
#	codex-rs/core/src/rollout/list.rs

pakrym-oai committed:
message_processor: enable active conversations in ConversationManager

aibrahim-oai reviewed (approved):
LGTM + minor ask for the test and struct/enum.

Admittedly, the filtering logic is ugly and needs revamping.

pakrym-oai committed:
Merge branch 'main' into pakrym/separate-interactive-and-non-interactive-sessions

# Conflicts:
#	codex-rs/app-server/src/message_processor.rs
#	codex-rs/exec/src/lib.rs
#	codex-rs/mcp-server/src/message_processor.rs

pakrym-oai commented on codex-rs/core/src/rollout/tests.rs:
```diff
@@ -754,3 +769,48 @@ async fn test_stable_ordering_same_second_pagination() {
 }
+
+#[tokio::test]
+async fn test_interactive_filter_excludes_non_interactive_sessions() {
```
The parameter name is interactive_only so the reverse should return both, I'll update the test.

pakrym-oai replied:
The parameter name is interactive_only so the reverse should return both, I'll update the test.

pakrym-oai committed:
rollout/tests: relax hardcoded session path checks in interactive filter test

Replace strict path equality assertions with filename-based checks to avoid dependency on exact directory structure. This makes the test more robust to changes in session path generation. Also update assertions for session counts accordingly.

pakrym-oai commented on codex-rs/core/src/rollout/recorder.rs:
```diff
@@ -89,8 +95,9 @@ impl RolloutRecorder {
         codex_home: &Path,
         page_size: usize,
         cursor: Option<&Cursor>,
+        interactive_only: bool,
```
Just for this parameter of for all of them? I think struct makes sense when we get more filters.

pakrym-oai committed:
use SessionSource in session listing and metadata

Introduce SessionSource to indicate the origin of a session (Cli, VSCode, Exec, Mcp, Unknown) instead of the interactive boolean. Update session metadata and listing/filtering to use SessionSource, including in ConversationManager, RolloutRecorder, apps, and tests. Adjust list_conversations APIs and migration points to accept allowed_sources for filtering, and update all callsites and test factories accordingly. Also, update serialization and default logic for SessionMeta and

pakrym-oai committed:
core/rollout: add INTERACTIVE_SESSION_SOURCES constant, use consistently in session filtering

- Define INTERACTIVE_SESSION_SOURCES (&[SessionSource::Cli, SessionSource::Vscode]) in core/rollout, export for reuse.
- Replace local INTERACTIVE_SOURCES constants/usages in core and tui, and update tests/production code to use INTERACTIVE_SESSION_SOURCES.
- Fix enum variant name VSCode -> Vscode for consistency.
- Update sample session creation in tests to specify

pakrym-oai committed:
protocol: consistently use VSCode variant instead of Vscode for SessionSource

Update naming of SessionSource::Vscode to SessionSource::VSCode everywhere for consistency with protocol definition, including in tests and session source filters.

aibrahim-oai commented on codex-rs/protocol/src/protocol.rs:
```diff
@@ -916,14 +916,42 @@ impl InitialHistory {
+pub enum SessionSource {
+    Cli,
+    #[default]
+    VSCode,
```
how about app-server instead of vscode?

pakrym-oai replied:
vscode is the name of the product, I'd like external customers to be able to understand what we're talking about in their session file.

aibrahim-oai replied:
It's internal code tho. Also, vscode may confuse other what about cursor and windsurf. I think people reading the code will have easier time referring to a crate/dir name

aibrahim-oai commented on codex-rs/protocol/src/protocol.rs:
```diff
@@ -916,14 +916,42 @@ impl InitialHistory {
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, Eq, TS, Default)]
+#[serde(rename_all = "snake_case")]
+pub enum SessionSource {
+    Cli,
```
how about `tui` because we have a `cli` crate

pakrym-oai replied:
codex cli is the name of the product

aibrahim-oai replied:
It's confusing with [`cli.rs`](https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci81OWExODBkZGVjNGFkYWY5NzYwOTcyY2RiMWViODlmMDZhODFiZThiL2YvY29kZXgtcnMvZXhlYy9zcmMvY2xpLnJzP3VybD1odHRwcyUzQSUyRiUyRmdpdGh1Yi5jb20lMkZvcGVuYWklMkZjb2RleCUyRiZsaW5lcz0xLTExNA%3D%3D?origin=gitlens) tho. I feel it's not clear which one to me.

aibrahim-oai commented on codex-rs/core/src/rollout/list.rs:
```diff
@@ -196,6 +201,13 @@ async fn traverse_directories_for_paths(
                     let summary = read_head_and_tail(&path, HEAD_RECORD_LIMIT, TAIL_RECORD_LIMIT)
                         .await
                         .unwrap_or_default();
+                    if !allowed_sources.is_empty()
```
clean

pakrym-oai committed:
protocol: change SessionSource enum renaming to "lowercase" for serde and typeshare

pakrym-oai commented:
Tested in CLI and VSCode.
